packages/@aws-cdk/aws-ecs/lib/task-definition.ts

```diff
@@ -0,0 +1,260 @@
+
+        return {
+            name: definition.name,
+            image: definition.image,
+            command: definition.command,
+            cpu: definition.cpu,
+            disableNetworking: definition.disableNetworking,
+            dnsSearchDomains: definition.dnsSearchDomains,
+            dnsServers: definition.dnsServers,
+            // dockerLabels: definition.dockerLabels,
+            dockerSecurityOptions: definition.dockerSecurityOptions,
+            entryPoint: definition.entryPoint,
+            essential: definition.essential,
+            hostname: definition.hostname,
+            links: definition.links,
+            // logConfiguration: logConfigs, // only set if passed in?
+            memory: definition.memory,
+            memoryReservation: definition.memoryReservation,
+            privileged: definition.privileged,
+            readonlyRootFilesystem: definition.readonlyRootFilesystem,
+            user: definition.user,
+            workingDirectory: definition.workingDirectory
+        };
+    }
+
+    private renderPlacementConstraint(pc: PlacementConstraint): cloudformation.TaskDefinitionResource.TaskDefinitionPlacementConstraintProperty {
+        return {
+            type: pc.type,
+            expression: pc.expression
+        };
+    }
+
+    // private renderLogConfiguration(lc: LogConfiguration[]): cloudformation.TaskDefinitionResource.ContainerDefinitionProperty.LogConfiguration[] {
+    //     return {
+    //         logDriver: lc.logDriver,
+    //         options: lc.options
+    //     };
+    // }
+
+    // private renderVolume(volume: Volume): cloudformation.TaskDefinitionResource.VolumeProperty {
+    //     return {
+    //         host: this.renderHost(volume.host),
+    //         name: volume.name
+    //     };
+    // }
+
+    // private renderHost(host: Host): cloudformation.TaskDefinitionResource.VolumeProperty.Host {
+    //     return {
+    //         sourcePath: host.sourcePath
+    //     }
+    // }
+}
+
+export interface ContainerDefinition {
+    name: string;
+    image: string;
+
+    command?: string[];
+    cpu?: number;
+    disableNetworking?: boolean;
+    dnsSearchDomains?: string[];
+    dnsServers?: string[];
+    dockerLabels?: string[];
+    dockerSecurityOptions?: string[];
+    entryPoint?: string[];
+    essential?: boolean;
+    hostname?: string;
+    links?: string[];
+    logConfiguration: LogConfiguration[];
+    memory?: number;
+    memoryReservation?: number;
+    privileged?: boolean;
+    readonlyRootFilesystem?: boolean;
+    user?: string;
+    workingDirectory?: string
+}
+// environment?: list of key-value;
+// extraHosts?: hostEntry[];
+// healthCheck?: healthCheck;
+// linuxParameters: linuxParam[];
+// mountPoints?: mountPoint[];
+// portMappings?: portMapping[];
+// ulimits?: ulimit[];
+// volumesFrom?: volumeFrom[];
+
+export interface PlacementConstraint {
```

Comment:
Does it make sense to model these as individual classes that inherit from `abstract class PlacementConstraint` ?